File: share/themes/default/styles.lua

--[[
	This Lua script is the stylesheet for the default theme.
	
	Note that all of the styles are commented out.
	This is because we're actually using the built-in defaults.
	
	Think of this as a guide to what the original values are when making your
	own themes.
]]--

-- This is a workaround for a bug when rendering in software mode.
local WHITE = 0xfffefefe

--[[
styles.body_font:set { size = 30, style = Font.PLAIN }
styles.body_fg = 0xffbfbfbf
styles.body_head_font:set { size = 30, style = Font.PLAIN }
styles.body_head_fg = WHITE
styles.body_aside_font:set { size = 20, style = Font.PLAIN }
styles.body_aside_fg = 0xffbfbfbf

styles.announcement_head_font:set { size = 30, style = Font.PLAIN }
styles.announcement_head_fg = WHITE
styles.announcement_body_font:set { size = 20, style = Font.PLAIN }
styles.announcement_body_fg = 0xff7f7f7f
styles.announcement_symbol_fg = 0xbfffffff
styles.announcement_bg = 0xbf000000

-- 100% HUD scale.
styles.hud_normal_font:set { size = 30, style = Font.BOLD }
styles.hud_normal_head_font:set { size = 20, style = Font.PLAIN }
-- 50% HUD scale.
styles.hud_small_font:set { size = 20, style = Font.BOLD }
styles.hud_small_head_font:set { size = 15, style = Font.PLAIN }

styles.console_font:set { name = "monospace", size = 30, style = Font.PLAIN }
styles.console_fg = WHITE
styles.console_cursor_fg = 0xffbfbfbf
styles.console_bg = 0xbf000000

styles.form_font = styles.body_font
styles.form_fg = WHITE
styles.form_disabled_fg = 0x7fffffff

styles.dialog_bg = 0xcc000000

styles.grid_margin.x = 6
styles.grid_margin.y = 6
styles.grid_padding.x = 1
styles.grid_padding.y = 1

styles.button_bg = 0x3f00007f
styles.button_disabled_bg = 0x3f7f7f7f
styles.button_pressed_bg = 0x7f00007f

styles.heading_font:set { size = 40, style = Font.BOLD }
styles.heading_fg = WHITE
]]--

File: engine/Display/Styles.cpp

// Styles.cpp
//
// Copyright (c) 2015 Michael Imamura.
//
// Licensed under GrokkSoft HoverRace SourceCode License v1.0(the "License");
// you may not use this file except in compliance with the License.
//
// A copy of the license should have been attached to the package from which
// you have taken this file. If you can not find the license you can not use
// this file.
//
//
// The author makes no representations about the suitability of
// this software for any purpose.  It is provided "as is" "AS IS",
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
// implied.
//
// See the License for the specific language governing permissions
// and limitations under the License.

#include "../Util/Config.h"

#include "Styles.h"

using namespace HoverRace::Util;

namespace HoverRace {
namespace Display {

Styles::Styles() :
	gridMargin(0, 0), gridPadding(0, 0)
{
	Reload();
}

void Styles::Reload()
{
	const Config *cfg = Config::GetInstance();
	const std::string &monospaceFontName = cfg->GetDefaultMonospaceFontName();

	// Hack fix for broken alpha blending when HW acceleration is off.
	// Only full-brightness, fully-opaque white is affected.
	const Color WHITE(cfg->runtime.noAccel ? 0xfffefefe : COLOR_WHITE);

	bodyFont.Set(30, 0);
	bodyFg = 0xffbfbfbf;
	bodyHeadFont.Set(30, 0);
	bodyHeadFg = WHITE;
	bodyAsideFont.Set(20, 0);
	bodyAsideFg = 0xffbfbfbf;

	announcementHeadFont.Set(30);
	announcementHeadFg = WHITE;
	announcementBodyFont.Set(20);
	announcementBodyFg = 0xff7f7f7f;
	announcementSymbolFg = 0xbfffffff;
	announcementBg = 0xbf000000;

	// 100% HUD scale.
	hudNormalFont.Set(30, UiFont::BOLD);
	hudNormalHeadFont.Set(20, 0);
	// 50% HUD scale.
	hudSmallFont.Set(20, UiFont::BOLD);
	hudSmallHeadFont.Set(15, 0);

	consoleFont.Set(monospaceFontName, 30);
	consoleFg = WHITE;
	consoleCursorFg = 0xffbfbfbf;
	consoleBg = 0xbf000000;

	formFont = bodyFont;
	formFg = WHITE;
	formDisabledFg = 0x7fffffff;

	dialogBg = 0xcc000000;

	gridMargin.x = 6;
	gridMargin.y = 6;
	gridPadding.x = 1;
	gridPadding.y = 1;

	buttonBg = 0x3f00007f;
	buttonFocusedBg = 0xff00007f;
	buttonDisabledBg = 0x3f7f7f7f;
	buttonPressedBg = 0x7f00007f;

	headingFont.Set(40, UiFont::BOLD);
	headingFg = WHITE;
}

}  // namespace Display
}  // namespace HoverRace
